code/modules/library/lib_items.dm

```diff
@@ -139,6 +139,14 @@
 			I.forceMove(get_turf(src))
 	..()
 
+/obj/structure/bookcase/cat_act(mob/M)
+	visible_message("<span class='danger'>\The [M] sharpens [M.gender == FEMALE ? "her" : "his"] claws on \the [src].</span>")
+	playsound(src, "paper_tear.ogg", 50, 1, -5)
+	shake(1,3)
+	for(var/obj/item/I in contents)
+		if(is_type_in_list(I, valid_types))
+			I.forceMove(get_turf(src))
```

Comment:
I don't see an implementation of `bookcase/Exited()` that would update its icon when the contents are changed, so I think you should call `update_icon()` after this.